Add a new note a day after adding the first product

moon0326 opened the pull request:
Fixes #5956 

This PR adds a new note with a link to the customizer doc a day after adding the first product.

### Screenshots

![Screen Shot 2021-01-08 at 10 15 42 AM](https://user-images.githubusercontent.com/4723145/104049893-974ab080-519a-11eb-83de-16ad52be8081.jpg)

### Detailed test instructions:

Make sure your store is less than 14 days old. If it has been active for more than 14 days, run the following query.

```update wp_options set option_value = UNIX_TIMESTAMP() where option_name='woocommerce_admin_install_timestamp'```

1. Start with a fresh installation of WooCommerce or delete all published products.
2. Add a new product.
3. Open `wc_posts` and find the product you just added in step #2. Update the `post_date` and `post_date_gmt` to the past date (at least -1 day)
2. Install & activate "WP Crontrol" plugin
3. Navigate to Tools -> Cron Events and run `wc_admin_daily` job
4. Navigate to WooCommerce -> Home and confirm a new note with the title "How to customize your product catalog"

Commit:
Add a new note a day after adding the first product

adrianduffell reviewed (dismissed):
Overall the code looks good 👍  Some tests for `get_note()` would be nice however.

I also found the note is triggered for established stores with existing products, but the advice in the note seems unwanted for those stores.

Perhaps there should be another check to determine if the store is new or established? I think you could use the store’s order count to do this. For example, if the store has existing orders, then don't display the note.

moon0326 commented:
> Overall the code looks good 👍 Some tests for `get_note()` would be nice however.
> 
> I also found the note is triggered for established stores with existing products, but the advice in the note seems unwanted for those stores.
> 
> Perhaps there should be another check to determine if the store is new or established? I think you could use the store’s order count to do this. For example, if the store has existing orders, then don't display the note.

Thank you for catching it 👍  I think that should be fixed. I will push a fix with a test.

I'm also going to see if this PR can use the Remote Inbox Notification system instead.

Commit:
Do not add note if store has been active for more than 14 days

moon0326 commented:
@adrianduffell Hey Adrian, I asked Pedro about the existing store issue. This is his reply: https://github.com/woocommerce/woocommerce-admin/issues/5956#issuecomment-760085082

tl;dr -- the store is active for less than 14 days.

I've updated the logic and added tests.

louwie17 commented on src/Events.php:
```diff
@@ -140,6 +141,12 @@ protected function possibly_add_notes() {
 		AddingAndManangingProducts::possibly_add_note();
+		CustomizingProductCatalog::possibly_add_note();
+
+		if ( $this->is_remote_inbox_notifications_enabled() ) {
+			DataSourcePoller::read_specs_from_data_sources();
+			RemoteInboxNotificationsEngine::run();
+		}
```
I believe this isn't necessary, as it's already being done above where `possible_add_notes` is being triggered.

https://github.com/woocommerce/woocommerce-admin/pull/6032/files#diff-2e1f1cedde54c64bfd2d2d1b61019c13af57b9c5e1dc1315d4c2dcc492ad66e1R96-R99

moon0326 replied:
Thank you @louwie17 I'm not sure why that those lines are highlighted as new. The last four lines are existing code for the Remote Inbox Engine. I think we can ignore them.

louwie17 replied:
@moon0326 are you sure? they don't show up on the `main` branch: https://github.com/woocommerce/woocommerce-admin/blob/main/src/Events.php#L142

louwie17 replied:
Thanks @moon0326, approving this :)

louwie17 commented on tests/notes/class-wc-tests-customizing-product-catalog.php:
```diff
@@ -0,0 +1,131 @@
+<?php
```
Nice work on these test cases, very thorough 👍 👍

moon0326 replied:
@louwie17 thank you!

louwie17 reviewed (changes requested):
Tested the store being older then 14 day's and the note didn't show 👍 also nice work on the tests.
I left one comment in the `CustomizingProductCatalog.php`, I believe calling the `DataSourcePoller` there isn't necessary (see comment).

Commit:
Remove remote inbox engine code -- added by accident

louwie17 reviewed (approved):
LGTM 🚀

moon0326 commented:
> LGTM 🚀

Thank you! I need your approval please 😄

louwie17 commented:
> Thank you! I need your approval please 😄

Ooh shoot haha you might need @adrianduffell approval, or you could dismiss his review, if you can (I haven't run into this before):
<img width="900" alt="Screen Shot 2021-01-20 at 10 03 06 PM" src="https://user-images.githubusercontent.com/2240960/105270071-81aa9300-5b6b-11eb-9287-188e48411492.png">

Commit:
Merge branch 'main' into add/5956-add-note-customizing-product-catalog

moon0326 commented:
> > Thank you! I need your approval please 😄
> 
> Ooh shoot haha you might need @adrianduffell approval, or you could dismiss his review, if you can (I haven't run into this before):
> <img alt="Screen Shot 2021-01-20 at 10 03 06 PM" width="900" src="https://user-images.githubusercontent.com/2240960/105270071-81aa9300-5b6b-11eb-9287-188e48411492.png">

Oh nice! Did not know that was a feature :) Thank you!
